DevCycle/DVCClient.swift

```diff
@@ -88,41 +88,54 @@ public class DVCClient {
             return
         }
         self.service = service
-        self.service?.getConfig(user: user, enableEdgeDB: self.enableEdgeDB, extraParams: nil, completion: { [weak self] config, error in
-            guard let self = self else { return }
-            if let error = error {
-                Log.error("Error getting config: \(error)", tags: ["setup"])
-                self.cache = self.cacheService.load()
-            } else {
-                if let config = config {
-                    Log.debug("Config: \(config)", tags: ["setup"])
-                }
-                self.config?.userConfig = config
-                
-                self.handleCachedAnonUserId(user: user)
+        
+        var cachedConfig: UserConfig?
+        if let configCacheTTL = options?.configCacheTTL, let disableConfigCache = options?.disableConfigCache, !disableConfigCache {
+            cachedConfig = cacheService.getConfig(user: user, ttlMs: configCacheTTL)
+        }
+        
+        if cachedConfig != nil {
+            self.config?.userConfig = cachedConfig
+            self.isConfigCached = true
+            Log.debug("Loaded config from cache")
+        } else {
+            self.service?.getConfig(user: user, enableEdgeDB: self.enableEdgeDB, extraParams: nil, completion: { [weak self] config, error in
+                guard let self = self else { return }
+                if let error = error {
+                    Log.error("Error getting config: \(error)", tags: ["setup"])
+                    self.cache = self.cacheService.load()
+                } else {
+                    if let config = config {
+                        Log.debug("Config: \(config)", tags: ["setup"])
+                    }
+                    self.config?.userConfig = config
+                    self.isConfigCached = false
```

Comment:
could we move this to be set to false any time a config is fetched? as opposed to just in the init

Reply:
Not sure how to set this value within the `RequestConsolidator` (where config is fetched) but I can also set it for `identify`, `reset`?